### PR TITLE
fixes #14594 - remove all templates from Katello

### PR DESCRIPTION
**Pull request by stbenjam**
The Foreman templates will support all our needs and we no longer need to maintain special snowflake Katello ones.

These PR's are prerequisites before merging this one:

- [x] Consolidate kickstart template https://github.com/theforeman/community-templates/pull/462
- [x] Refactor redhat_register snippet https://github.com/theforeman/community-templates/pull/302
- [x] Update foreman https://github.com/theforeman/foreman/pull/5312
- [x] Sync updated redhat_register to foreman so nightly provisioning works

### CODE REVIEW

**Comment by stbenjam** [comment]
[test katello]

**Comment by jturel** [comment]
@stbenjam can you summarize what a good testing strategy for this would be? I haven't gotten into the provisioning side too much and want to make sure I don't miss something obvious

**Comment by jlsherrill** [comment]
@jturel that is a very loaded question :) 

If you want, I'm happy to run down some of the provisioning stuff and we can review this together!

**Comment by stbenjam** [comment]
Yes it's rather challenging to go through the whole provisioning dance, it sounds like @jlsherrill will help - if you need me for anything let me know

**Comment by jturel** [comment]
@stbenjam have you tested this end to end, through to actually provisioning system?

**Comment by stbenjam** [comment]
I have, I used the centos7-provisioning-nightly box with all the linked PR's applied and provisioned a host with content from Katello.

**Comment by stbenjam** [comment]
Thanks!
